generate-stats

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import ssl
+import sys
+import sqlite3
+import urllib.request
+import argparse
+import logging
+import re
+import json
+
+from task import github
+from machine.machine_core.constants import IMAGES_DIR
+
+def main():
+    parser = argparse.ArgumentParser(description='Store information about failed tests')
+    parser.add_argument("--db", default="cockpit-failures.db", help="Database name")
+    parser.add_argument("--verbose", action="store_true", help="Enable verbose logging")
+    opts = parser.parse_args()
+
+    if opts.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    db_conn = sqlite3.connect(opts.db)
+    cursor = db_conn.cursor()
+
+    all_tables = []
+
+    # Get all failures sorted by number of occurrences
+    top10 = []
+    rows = cursor.execute("select testname, COUNT(run) from Failures GROUP BY testname ORDER BY COUNT(run) DESC LIMIT 10;").fetchall()
```

Comment:
I think we should limit this by default to the last 7 days.  Otherwise the "sins of the past" will haunt us forever, and we won't see the effect of test fixes.

Reply:
Well, we don't have any timestamp in the DB. We can either add it or do different limitation like "last 1000 failed tests".

Reply:
Sent https://github.com/cockpit-project/bots/pull/1034, lets block this for a bit and then use with timestamp.